client/src/components/relatedProducts/test/relatedProducts.test.js

```diff
@@ -16,9 +16,12 @@ describe('relatedProducts', () => {
   );
 
   it('should render RelatedProducts component without crashing', () => {
+    const location = {
+      pathname: 47421,
+    };
     const { getByTestId } = render(
       <Provider store={testStore}>
-        <RelatedProductsContainer productId={47421}/>
+        <RelatedProductsContainer productId={47421} />
```

Comment:
```suggestion
        <RelatedProductsContainer productId='47421' />
```
Looks like your prop type should be a string!

Reply:
Also regarding the duplicate results from API, maybe we need to filter that and only keep unique ones in the future?